server/controllers/chartsController/helpers.js

```diff
@@ -303,3 +331,84 @@ export const calculateSubjectVariableDayCount = (
 
   return subjectsDayDataAndVariable.length
 }
+
+export const addSubjectsToResultsFromFiltersMap = ({
+  subjectsInFiltersA,
+  subjectsInFiltersB,
+  mapForResultsFromFilters,
+}) => {
+  subjectsInFiltersA.filter((subjectData) => {
+    const foundSubject = subjectsInFiltersB.find(
+      ({ subject }) => subject === subjectData.subject
+    )
+    const isSubjectInMap = mapForResultsFromFilters?.has(subjectData.subject)
+
+    if (foundSubject?.subject) {
+      if (isSubjectInMap) {
+        const subjectDataFromMap = mapForResultsFromFilters.get(
```

Comment:
what's the difference between `subjectDataFromMap` and `foundSubject`? Also, instead of `get`ting the subject from the map here, and checking if the map `has` the subject above, what if we `get` the subject above and use that to determine `isSubjectInMap`?

Reply:
https://github.com/AMP-SCZ/dpdash/pull/280/commits/c30a7d514c931ad36f83916c440ff7812449f315